File: frontend/src/types/turn.ts
export type GamePhase =
  | "INIT"
  | "MAYOR_CAMPAIGN"
  | "MAYOR_VOTE"
  | "NIGHT"
  | "DAY_BID"
  | "DAY_SPEECH"
  | "VOTE"
  | "GAME_OVER";

export type DeceptionLabel =
  | "truthful"
  | "omission"
  | "distortion"
  | "fabrication"
  | "misdirection";

export interface Turn {
  id: string;
  game_id: string;
  player_id: string;
  round_number: number;
  phase: GamePhase;
  prompt_sent: string | null;
  completion_received: string | null;
  private_reasoning: string | null;
  public_statement: string | null;
  vote_target: string | null;
  bid_level: number | null;
  technique_self_label: string | null;
  deception_self_label: DeceptionLabel | null;
  confidence: number | null;
  is_default_response: boolean;
  token_count_input: number | null;
  token_count_output: number | null;
  latency_ms: number | null;
  created_at: string;
}

File: backend/app/schemas/turn.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TurnResponse(BaseModel):
    """Response schema for a single turn."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    game_id: uuid.UUID
    player_id: uuid.UUID
    round_number: int
    phase: str
    prompt_sent: str | None = None
    completion_received: str | None = None
    private_reasoning: str | None = None
    public_statement: str | None = None
    vote_target: uuid.UUID | None = None
    bid_level: int | None = None
    technique_self_label: str | None = None
    deception_self_label: str | None = None
    confidence: int | None = None
    is_default_response: bool
    token_count_input: int | None = None
    token_count_output: int | None = None
    latency_ms: int | None = None
    created_at: datetime
